persist/persist_test.go

```diff
@@ -26,3 +26,71 @@ func TestIntegrationRandomSuffix(t *testing.T) {
 		file.Close()
 	}
 }
+
+// TestUnitNewSafeFile checks that a new file is created and that its name is
+// different than the finalName of the new safeFile.
+func TestUnitNewSafeFile(t *testing.T) {
+	// Create safe file.
+	sf, err := NewSafeFile("NewSafeFile test file" + RandomSuffix())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(sf.Name())
+	defer sf.Close()
+
+	// Check that the name of the file is not equal to the final name of the
+	// file.
+	if sf.Name() == sf.finalName {
+		t.Errorf("safeFile temporary filename and finalName are equivalent: %s\n", sf.Name())
+	}
```

Comment:
This test seems awkward and implementation dependent.

The input string to `NewSafeFile` is what the final name should be, right? So

```go
finalName := filepath.Join(build.SiaTestingDir, persistDir, "Test...")
NewSafeFile(finalName)
// more stuff
if sf.Name() == finalName {
// more stuff
sf.Commit()
os.Open(finalName)
// check that the contents of the file are correct
```